js/memory.js

```diff
@@ -1,9 +1,40 @@
+
 class MemoryGame {
-  constructor(cards){
+  constructor(cards, pickedCards,pairsClicked, pairsGuessed){
     this.cards = cards;
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
     // add the rest of the class properties here
   }
-  shuffleCards() {}
-  checkIfPair(card1, card2) {}
-  isFinished() {}
+  shuffleCards(cards) {
+    if (!cards){
+      return undefined;
+    }
+    var currentIndex = this.cards.length, temporaryValue, randomIndex;
+    while (0 !== currentIndex) {
+    randomIndex = Math.floor(Math.random() * currentIndex);
+    currentIndex -= 1;
+    temporaryValue = this.cards[currentIndex];
+    this.cards[currentIndex] = this.cards[randomIndex];
+    this.cards[randomIndex] = temporaryValue;
+    }
+    return this.cards;
+  }
+  checkIfPair(card1, card2) {
+    this.pairsClicked =+ 1;
+    if (card1 === card2){
+      this.pairsClicked =+ 1;
+      this.pairsGuessed =+ 1;
+      return true;
+    }
+    return false;
+  }
+  isFinished() {
+    if (this.pairsClicked.length === (this.cards.length/2)){
```

Comment:
The condition should be checking for the number of guessed pairs `if (this.pairsClicked.length === this.cards.length/2){...}`

Also the way you wrote `isFinished` is fine however because the condition you have will already return as `true` or `false`, you can skip using the if statement altogether and return the result of the comparison
```
isFinished(){
  return this.pairsClicked.length === this.cards.length/2;
}